frontend/src/components/Profile.jsx

```diff
@@ -1,61 +1,171 @@
 import React, { useEffect, useState } from "react";
 import { auth, db } from "../config/firebase-config";
-import { doc, getDoc } from "firebase/firestore";
+import { doc, getDoc, setDoc, deleteDoc } from "firebase/firestore";
 
 function Profile() {
-  const [userDetails, setUserDetails] = useState(null);
-  const fetchUserData = async () => {
-    auth.onAuthStateChanged(async (user) => {
-      console.log(user);
-
-      const docRef = doc(db, "Users", user.uid);
-      const docSnap = await getDoc(docRef);
-      if (docSnap.exists()) {
-        setUserDetails(docSnap.data());
-        console.log(docSnap.data());
-      } else {
-        console.log("User is not logged in");
-      }
-    });
-  };
-  useEffect(() => {
-    fetchUserData();
-  }, []);
-
-  async function handleLogout() {
-    try {
-      await auth.signOut();
-      window.location.href = "/login";
-      console.log("User logged out successfully!");
-    } catch (error) {
-      console.error("Error logging out:", error.message);
+    const [major, setMajor] = useState("");
+    const [interests, setInterests] = useState([]);
+    const [userDetails, setUserDetails] = useState(null);
+    const [isEditing, setIsEditing] = useState(false);
+
+    const majorOptions = 
+    [
+        "Non-Engineering",
+        "Aerospace Engineering",
+        "Applied and Computational Mathematics",
+        "Applied Mechanics",
+        "Astronautical Engineering",
+        "Biomedical Engineering",
+        "Chemical Engineering",
+        "Civil Engineering",
+        "Computer Engineering and Computer Science",
+        "Computer Science",
+        "Computer Science Games",
+        "Computer Science & Business Administration",
+        "Electrical Engineering",
+        "Environmental Engineering",
+        "Industrial and Systems Engineering",
+        "Mechanical Engineering",
+        "Physics/Computer Science",
+        "Petroleum Engineering"
+    ];
+    
+    const fetchUserData = async () => {
+        auth.onAuthStateChanged(async (user) => {
+        
+            const docRef = doc(db, "Users", user.uid);
+            const docSnap = await getDoc(docRef);
+            if (docSnap.exists()) {
+                setUserDetails(docSnap.data());
+            }
+        });
+
+    };
+    
+    useEffect(() => {
```

Comment:
what does this `useEffect` do? i'm assuming it runs the block as soon as this webpage loads?

Reply:
Yes that is it